src/azul/chalice.py

```diff
@@ -259,13 +263,23 @@ def _security_headers_middleware(self, event, get_response):
         response.headers['Cache-Control'] = cache_control
         return response
 
-    def _retry_503(self, event, get_response):
+    def _retry_after(self, event, get_response):
         """
-        Add a retry-after header to 503 responses
+        Add a retry-after header to the response based on the response type.
         """
         response = get_response(event)
-        if response.status_code == 503:
+        status = response.status_code
+        if status in [307, 503]:
             response.headers.setdefault('Retry-After', '30')
+        if status == 307:
+            # When a client's request fails due to an internal request to TDR
+            # timing out, we want the client to retry their request, so we
+            # return a 307 response along with a location header containing the
+            # URL the client had just requested.
+            location = furl(self.base_url,
+                            path=event.context['path'],
+                            args=event.query_params)
+            response.headers.setdefault('Location', location.url)
```

Comment:
```suggestion
            response.headers.setdefault('Location', str(location))
```

is what we typically use.

Reply:
It's also a bit odd that we compute the value only to discard it if the header is already present.